### PR TITLE
translate-c: add <assert.h> support

### PR DESCRIPTION
**Pull request by ehaas**
Implement __builtin_expect so C code that uses assert() can be translated.

### CODE REVIEW

**Review by andrewrk — APPROVED** [review]

**Comment by ehaas** [comment]
Looks like assert on windows is doing something else...I can investigate it over the weekend.

**Comment by ehaas** [comment]
`assert()` on windows uses wide characters so I'll need to add support for that before this will work. Will do as separate PRs.

**Comment by ehaas** [comment]
Currently blocked on #7989 7989

**Comment by ehaas** [comment]
The remaining problem with this branch (Linux assert.h) should be fixed once the `translate-c-ast` branch is merged, due to improved if/else handling.

**Pull request merged**
